habitat_baselines/rl/ddppo/policy/resnet_policy.py

```diff
@@ -234,33 +231,32 @@ def __init__(
         backbone,
         resnet_baseplanes,
         normalize_visual_inputs: bool,
-        fuse_keys: Optional[List[str]],
         force_blind_policy: bool = False,
         discrete_actions: bool = True,
     ):
         super().__init__()
         self.prev_action_embedding: nn.Module
         self.discrete_actions = discrete_actions
+        self._n_prev_action = 32
         if discrete_actions:
-            self.prev_action_embedding = nn.Embedding(action_space.n + 1, 32)
+            self.prev_action_embedding = nn.Embedding(
+                action_space.n + 1, self._n_prev_action
+            )
         else:
             num_actions = get_num_actions(action_space)
-            self.prev_action_embedding = nn.Linear(num_actions, 32)
+            self.prev_action_embedding = nn.Linear(
+                num_actions, self._n_prev_action
+            )
 
-        self._n_prev_action = 32
         rnn_input_size = self._n_prev_action  # test
 
         # Only fuse the 1D state inputs. Other inputs are processed by the
         # visual encoder
-        self._fuse_keys: List[str] = (
-            [
-                k
-                for k in fuse_keys
-                if len(observation_space.spaces[k].shape) == 1
-            ]
-            if fuse_keys is not None
```

Comment:
The fuse_keys field is still needed. We have specific processing for a lot of observation types and this will duplicate that (plus break existing checkpoints).

Reply:
fuse_keys is in None of the configs. To me it acts as a duplicate of GYM.OBS_KEYS that also filters the observations keys. 
Do you have more information about these "do not break checkpoints" ? Where are they tested?

Reply:
I agree with Erik that this is still needed. In the current form, OBS_KEYS does function exactly the same. However, users should have the flexibility to use some subset of the observation keys in the policy input. For example, if someone wants to train a vision only policy, then they can use fuse_keys=[]. We should decouple the observation keys the environment exposes from the observation keys the policy takes as input.

Reply:
I'd rather us think about the use cases we have rather than the use cases we can imagine. Do we have an example of task where we want the environment to give and observation but not use it in the policy? If so, why can't it be a measure? fuse_keys is currently set from GYM.OBS_KEYS so the feature we are talking about does not even exist today! 
Regarding these DDPPO models, where are they tested? If they are not tested, I can't imagine they still work today.
I can put back fuse_keys, I don't really mind. But I want us to be very careful with the code we want to keep.

Reply:
I do know for a fact that I will someday forget to set `TASK_CONFIG.GYM.OBS_KEYS` because I will assume that the default when no `TASK_CONFIG.GYM.OBS_KEYS` is specified that all of the sensors are used (because that is how the gym wrapper works). Then I will be horrified when I realize 15 days and 20 Billion steps later that the policy ignored all of my 1D sensors (but not the visual ones). I feel like we have too much sensor filtering going on in my opinion. We specify the sensor list in the task, we specify a subset of `GYM.OBS_KEYS` is the task as well and finally, we have fuse_keys that we can overwrite in the policy config. 
At least let's make it so that the default is to use all of the 1D sensors instead of none of them?